test/tools/libtesteth/boostTest.cpp

```diff
@@ -137,6 +150,34 @@ int main(int argc, const char* argv[])
     int result = 0;
     auto fakeInit = [](int, char* []) -> boost::unit_test::test_suite* { return nullptr; };
     result = unit_test_main(fakeInit, argc, const_cast<char**>(argv));
+
+    // Print suggestions of a test case
+    if (result == boost::exit_exception_failure)  // test suite not found
+    {
+        size_t allTestsElementIndex = 0;
+        typedef std::pair<size_t, size_t> NameDistance;  // <index in availableTests, compared
+                                                         // distance>
+
+        // Use `vector` here because `set` does not work with sort
+        std::vector<NameDistance> distanceMap;  // <test name, compared distance>
+        for (auto& it : c_allTests)
+        {
+            int dist =
+                test::levenshteinDistance(sMinusTArg.c_str(), sMinusTArg.size(), it, strlen(it));
+            distanceMap.emplace_back(allTestsElementIndex++, dist);
+        }
+        std::sort(distanceMap.begin(), distanceMap.end(),
+            [](NameDistance const& _a, NameDistance const& _b) { return _a.second < _b.second; });
+        std::cerr << "Did you mean: \n";
+        std::set<string> suggestionSet;
+        for (size_t i = 0; i < 3 && i < distanceMap.size(); i++)
+        {
+            string name = c_allTests[distanceMap[i].first];
+            suggestionSet.emplace(name);
```

Comment:
Why do you need set here? It will contain fewer that 3 item if there are duplicates, and they will be ordered alphabetically instead of by distance when you output them.